browser/admin/src/integrator/AdminIntegratorSettings.ts

```diff
@@ -890,7 +894,14 @@ class SettingIframe {
 		if (!isDisabled) {
 			checkboxWrapper.addEventListener('click', () => {
 				onClickHandler(inputCheckbox, checkboxWrapper, materialIconContainer);
+				if (checkboxWrapper.id === 'Grid-ShowGrid-container') {
+					this.toggleGridOptionsVisibility(checkboxWrapper);
+				}
 			});
+			if (checkboxWrapper.id === 'Grid-ShowGrid-container') {
+				// Set the initial state of Grid fieldsets' visibility
+				setTimeout(() => this.toggleGridOptionsVisibility(checkboxWrapper), 0);
```

Comment:
we do we need this timeout?

Reply:
by this point, the DOM is not constructed. `setTimeout` will move this callback after the DOM is constructed.